apps/emqx_cluster_link/src/emqx_cluster_link_app.erl

```diff
@@ -32,31 +38,34 @@ prep_stop(State) ->
 stop(_State) ->
     _ = emqx_cluster_link:delete_hook(),
     _ = emqx_cluster_link:unregister_external_broker(),
-    _ = stop_msg_fwd_resources(emqx_cluster_link_config:links()),
+    _ = remove_msg_fwd_resources(emqx_cluster_link_config:links()),
     ok.
 
 %%--------------------------------------------------------------------
 %% Internal functions
 %%--------------------------------------------------------------------
 
-enabled_links() ->
-    lists:filter(
-        fun(#{enable := IsEnabled}) -> IsEnabled =:= true end,
-        emqx_cluster_link_config:links()
-    ).
-
-start_msg_fwd_resources(LinksConf) ->
+maybe_start_msg_fwd_resources(LinksConf) ->
     lists:foreach(
-        fun(LinkConf) ->
-            {ok, _} = emqx_cluster_link_mqtt:ensure_msg_fwd_resource(LinkConf)
+        fun(#{upstream := Name} = LinkConf) ->
+            %% This is mostly to make sure a new node joining the cluster works properly
+            %% if there is an active link.
+            case emqx_cluster_link_extrouter:list_actors(Name) of
```

Comment:
(Related to the my previous rant) Nit, but probably unrelated if operational model is simplified. From the design perspective, it's strange to see the implementation details of route replication affect this layer, basically acting as a configuration.

Reply:
Simplified: no more guesses and attempts to support half-closed links. If the link is enabled, the cluster tries to start its msg forwarding resource